test/integration/test/pagekite_access_test.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const spawn = require('child_process').spawn;
+const chakram = require('chakram'), expect = chakram.expect;
+const Config = require('config-js');
+
+var philipshue_server = require('../lib/philipsHue_server.js');
+var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
+var config = new Config('./test/integration/lib/config/foxbox.js');
+var header = new Config('./test/integration/lib/config/header.js');
+
+describe('Initiate the remote connection',function(){
+  var credential = config.get('credential'); 
+  var lightinfo; // to store the light status received from foxbox
+  var foxbox_process;
+
+  var setupURL = config.get('foxbox.url') + '/users/setup';
+  var loginURL = 'http://' + config.get('pagekite.externalURL')+ '/users/login';
+  var serviceURL = 'http://' + config.get('pagekite.externalURL')+ '/services/';
+  var serviceListURL = serviceURL + 'list';  
+  
+  before(function(done){
+
+    var FOXBOX_STARTUP_WAIT_TIME_IN_MS = 3000;
+    this.timeout(500000);
+
+    // start nupnp server
+    nupnp_server.start(config.get('nupnp_server.id'),
+      config.get('philips_hue.url') +':'+config.get('philips_hue.port'),
+      config.get('nupnp_server.port'));
+    
+    // configure philips hub simulator
+    philipshue_server.setup(config.get('philips_hue.port'));
+
+    // start foxbox
+    foxbox_process = spawn('./target/debug/foxbox', 
+      ['-c',  config.get('nupnp_server.param')+';'+ 
+      config.get('nupnp_server.url')+':'+ 
+      config.get('nupnp_server.port')+'/',
+      '-r',config.get('pagekite.r'),
+      '-t',config.get('pagekite.t'),
+      '-s',config.get('pagekite.s'),
+      '--remote-name',config.get('pagekite.externalURL'),
+      '--disable-tls'] );
+
+    // give time until foxbox is operational
+    setTimeout(done, FOXBOX_STARTUP_WAIT_TIME_IN_MS);  
+  });
+  
+  after(function(){
+    foxbox_process.kill('SIGINT');
+    require('../lib/foxboxHelper.js').removeDB();
+    
+    nupnp_server.stop();
+    philipshue_server.stop();
+  });
+  
+  // Theoretically, new account creation is not supported remotely
+  // TODO: a new test should be added once it is implemented
+  it('Create account locally',function(){
+
+    return chakram.post(setupURL, credential)
+    .then(function(setupResp){
+      expect(setupResp).to.have.status(201);
+    });
+  });
+
+  it('login to foxbox remotely',function(){
```

Comment:
I just realized the tests are similar to the first one uploaded. Maybe we could factorize them with something similar to:
http://stackoverflow.com/questions/29312331/how-do-i-factor-out-common-before-calls-with-mocha